app/controllers/api/automate_domains_controller.rb

```diff
@@ -72,5 +96,22 @@ def resource_search(id, type, klass)
     def current_tenant
       User.current_user.current_tenant || Tenant.default_tenant
     end
+
+    def valid_ref_type?(data)
+      return false unless data.key?("ref_type")
+      return true if data["ref_type"] == "tag" || data["ref_type"] == "branch"
+      false
```

Comment:
you can replace 101-103 with:
   %w(tag branch).include?(data["ref_type"]) if data.key?("ref_type")

Reply:
@abellotti Will do. Thank you.